test/cut/test_custom_attrs_randomized.py

```diff
@@ -0,0 +1,61 @@
+from hypothesis import given, reproduce_failure, settings
+from hypothesis import strategies as st
+
+from lhotse.array import seconds_to_frames
+from lhotse.testing.fixtures import RandomCutTestCase
+
+EXAMPLE_TIMEOUT_MS = 1000
+MAX_EXAMPLES = 500
+
+
+class TestCustomAttrPaddingRandomized(RandomCutTestCase):
+    @settings(deadline=EXAMPLE_TIMEOUT_MS, max_examples=MAX_EXAMPLES, print_blob=True)
+    @given(
+        st.one_of(
+            st.just(8000),
+            st.just(16000),
+            st.just(22050),
+            st.just(44100),
+            st.just(48000),
+        ),
+        st.one_of(st.just(160), st.just(200), st.just(256)),
+        st.one_of(st.just("left"), st.just("right"), st.just("both")),
+        st.data(),
+    )
+    def test_invariants_pad(
+        self, sampling_rate: int, window_hop: int, pad_direction: str, rand_gen
+    ):
+        # Generate cut duration in numbers of samples
+        num_samples = rand_gen.draw(
+            st.integers(round(sampling_rate * 0.46), round(sampling_rate * 1.9)),
+            label="Number of audio samples in a cut.",
+        )
+        # Generate random cut
+        frame_shift = window_hop / sampling_rate
+        cut = self.with_cut(
+            sampling_rate=sampling_rate,
+            num_samples=num_samples,
+            frame_shift=frame_shift,
+            features=False,
+            custom_field=True,
+        )
+        # Pad with random duration
+        duration = rand_gen.draw(
+            st.floats(
+                min_value=cut.duration + 0.03 * cut.duration, max_value=cut.duration * 2
+            ),
+            label=f"Padded cut duration",
+        )
+        padded = cut.pad(
+            duration=duration,
+            direction=pad_direction,
+            pad_value_dict={"codebook_indices": -1},
+        )
+        # Test the invariants
+        array = padded.load_codebook_indices()
```

Comment:
This randomized test runs 500 times with cuts of different sampling_rates, durations, etc. It attaches a 1D TemporalArray called "codebook_indices" with random ints, pads the cut, and reads out the padded array from the padded cut.